### PR TITLE
Block the tree on  mac_android test failures running on luci.

### PR DESCRIPTION
**Pull request by godofredoc**
Start blocking the tree on LUCI mac android failures. Mac android tests will now run exclusively on LUCI.

Bug: https://github.com/flutter/flutter/issues/74431

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by jmagman — APPROVED** [review]
LGTM

File: dev/prod_builders.json [hunk 1]
```diff
@@ -450,6 +450,240 @@
...
+         "name": "Mac_android android_plugin_example_app_build_test",
+         "repo": "flutter",
+         "task_name": "mac_android_android_plugin_example_app_build_test",
+         "flaky": false
+      },
+      {
+         "name": "Mac_android android_semantics_integration_test",
+         "repo": "flutter",
+         "task_name": "mac_android_android_semantics_integration_test",
+         "flaky": true
```
Comment (christopherfujino) [inline dev/prod_builders.json:463]: actually, can you mark this not flaky? this should pass at ToT

File: dev/prod_builders.json [hunk 2]
```diff
@@ -450,6 +450,240 @@
...
+         "name": "Mac_android home_scroll_perf__timeline_summary",
+         "repo": "flutter",
+         "task_name": "mac_android_home_scroll_perf__timeline_summary",
+         "flaky": false
+      },
+      {
+         "name": "Mac_android hot_mode_dev_cycle__benchmark",
+         "repo": "flutter",
+         "task_name": "mac_android_hot_mode_dev_cycle__benchmark",
+         "flaky": true
```
Comment (christopherfujino) [inline dev/prod_builders.json:589]: this should also not be flaky, as it should pass after a recent fix

**Review by christopherfujino — CHANGES_REQUESTED** [review]

File: dev/prod_builders.json (hunk 1 above)
Comment (godofredoc) [reply dev/prod_builders.json:463]: done

File: dev/prod_builders.json (hunk 2 above)
Comment (christopherfujino) [reply dev/prod_builders.json:589]: https://github.com/flutter/flutter/commit/75dc45aba388b6db05914ae1d20c9298a0191172 should have fixed this one.

File: dev/prod_builders.json [hunk 3]
```diff
@@ -450,6 +450,240 @@
...
+         "name": "Mac_android integration_ui_textfield",
+         "repo": "flutter",
+         "task_name": "mac_android_integration_ui_textfield",
+         "flaky": false
+      },
+      {
+         "name": "Mac_android microbenchmarks",
+         "repo": "flutter",
+         "task_name": "mac_android_microbenchmarks",
+         "flaky": true
```
Comment (christopherfujino) [inline dev/prod_builders.json:631]: can you mark this as not flaky? I believe this is a recent regression.

File: dev/prod_builders.json (hunk 2 above)
Comment (godofredoc) [reply dev/prod_builders.json]: done

File: dev/prod_builders.json (hunk 3 above)
Comment (godofredoc) [reply dev/prod_builders.json]: done

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**